abmarl/sim/gridworld/wrapper.py

```diff
@@ -0,0 +1,169 @@
+        """
+        return self.wrapped_component.grid
+
+    @abstractmethod
+    def check_space(self, space):
+        """
+        Verify that the space can be wrapped.
+        """
+        pass
+
+    @abstractmethod
+    def wrap_space(self, space):
+        """
+        Wrap the space.
+
+        Args:
+            space: The space to wrap.
+        """
+        pass
+
+    @abstractmethod
+    def wrap_point(self, space, point):
+        """
+        Wrap a point to the space.
+
+        Args:
+            space: The space into which to wrap the point.
+            point: The point to wrap.
+        """
+        pass
+
+
+class ActorWrapper(ComponentWrapper, ActorBaseComponent):
+    """
+    Wraps an ActorComponent.
+
+    Modify the action space of the agents involved with the Actor, namely the specific
+    actor's channel. The actions recieved from the trainer are the wrapped space,
+    so we need unwrap them to send them to the actor. This is the opposite from
+    how we wrap and unwrap observations.
+    """
+    def __init__(self, component):
+        assert isinstance(component, ActorBaseComponent), \
+            "Wrapped component must be an ActorBaseComponent."
+        self._actor = component
+        # Need to record the pre-wrapped space for the wrapping functions.
+        self.from_space = {
+            agent.id: agent.action_space[self.key]
+            for agent in self.agents.values()
+            if isinstance(agent, self.supported_agent_type)
+        }
+        for agent in self.agents.values():
+            if isinstance(agent, self.supported_agent_type):
+                assert self.check_space(agent.action_space[self.key]), \
+                    f"Cannot wrap {self.key} action channel for agent {agent.id}"
+                agent.action_space[self.key] = self.wrap_space(agent.action_space[self.key])
+
+    @property
+    def wrapped_component(self):
+        """
+        Get the wrapped actor.
+        """
+        return self._actor
+
+    @property
+    def key(self):
+        """
+        The key is the same as the wrapped actor's key.
+        """
+        return self.wrapped_component.key
+
+    @property
+    def supported_agent_type(self):
+        """
+        The supported agent type is the same as the wrapped actor's supported agent type.
+        """
+        return self.wrapped_component.supported_agent_type
+
+    def process_action(self, agent, action_dict, **kwargs):
+        """
+        Unwrap the action and pass it to the wrapped actor to process.
+
+        Args:
+            agent: The acting agent.
+            action_dict: The action dictionary for this agent in this step. The
+                action in this channel comes in the wrapped space.
+        """
+        if isinstance(agent, self.supported_agent_type):
+            action = action_dict[self.key]
+            wrapped_action = self.wrap_point(self.from_space[agent.id], action)
```

Comment:
Should change to `unwraped_action` for clarity, even though we use the function `wrap_point`.